app/src/main/java/com/keylesspalace/tusky/fragment/NotificationsFragment.kt

```diff
@@ -0,0 +1,688 @@
+                                is StatusActionSuccess.Reblog ->
+                                    statusViewData.status.copy(reblogged = it.action.state)
+                                is StatusActionSuccess.VoteInPoll ->
+                                    statusViewData.status.copy(
+                                        poll = it.action.poll.votedCopy(it.action.choices)
+                                    )
+                            }
+                            indexedViewData.value?.statusViewData = statusViewData.copy(
+                                status = status
+                            )
+
+                            adapter.notifyItemChanged(indexedViewData.index)
+                        }
+                }
+
+                // Refresh adapter on mutes and blocks
+                this.launch {
+                    viewModel.uiSuccess.collectLatest {
+                        when (it) {
+                            is UiSuccess.Block, is UiSuccess.Mute, is UiSuccess.MuteConversation ->
+                                adapter.refresh()
+                            else -> { /* nothing to do */ }
+                        }
+                    }
+                }
+
+                // Update filter option visibility from uiState
+                this.launch {
+                    viewModel.uiState.collectLatest { updateFilterVisibility(it.showFilterOptions) }
+                }
+
+                // Update status display from statusDisplayOptions. If the new options request
+                // relative time display collect the flow to periodically re-bind the UI.
+                this.launch {
+                    viewModel.statusDisplayOptions
+                        .collectLatest {
+                            adapter.statusDisplayOptions = it
+                            layoutManager?.findFirstVisibleItemPosition()?.let { first ->
+                                first == RecyclerView.NO_POSITION && return@let
+                                val count = layoutManager!!.findLastVisibleItemPosition() - first
+                                adapter.notifyItemRangeChanged(
+                                    first,
+                                    count,
+                                    null
+                                )
+                            }
+
+                            if (!it.useAbsoluteTime) {
+                                updateTimestampFlow.collect()
+                            }
+                        }
+                }
+
+                // Update the UI from the loadState
+                adapter.loadStateFlow
+                    .distinctUntilChangedBy { it.refresh }
+                    .collect { loadState ->
+                        binding.recyclerView.isVisible = true
+                        binding.swipeRefreshLayout.isRefreshing =
+                            loadState.refresh is LoadState.Loading
```

Comment:
The swipe refresh spinner also shows up when blocking/muting/responding to follow requests, which feels a bit weird, but it is no big deal.